### PR TITLE
Doug Welchons - Cribs IC

### PR DESCRIPTION
**Pull request by DougWelchons**

### CODE REVIEW

File: lib/house.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
+class House
+  attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
```
Comment (mikedao) [inline lib/house.rb:5]: This should return an integer, not a string.

File: lib/house.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    total_area
+  end
+
+  def details
+    {"price" => price_to_intiger,
+     "address" => address}
+  end
+
+  def rooms_sorted_by_area
+    @rooms.sort_by {|room| room.area}.reverse
```
Comment (mikedao) [inline lib/house.rb:42]: Use the multi line syntax for enumerables given the choice

File: lib/house.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  end
+
+  def price_per_square_foot
+    (price_to_intiger / area.to_f).round(2)
+  end
+
+  def sort_bedrooms
+    @bedrooms = @rooms.select do |room|
```
Comment (mikedao) [inline lib/house.rb:50]: No need to set return value to an instance variable

**Review by mikedao — COMMENTED** [review]
The separate sorting was a bit needlessly complex. Make sure you are reading the specification closely.

Score: 3.5

**Comment by DougWelchons** [comment]
Thanks Mike! I realized about five minute after I submitted my PR that I had forgotten about my `rooms_from_category(type)` in my last minute panic to finish the IC, and this would have been a MUCH better option for the `rooms_by_category` method.

**Pull request closed without merging**